src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -10482,25 +10200,25 @@ instruct andL_rReg(rRegL dst, rRegL src, rFlagsReg cr)
 %}
 
 // And Register with Immediate 255
-instruct andL_rReg_imm255(rRegL dst, immL_255 src)
+instruct andL_rReg_imm255(rRegL dst, rRegL src, immL_255 mask)
 %{
-  match(Set dst (AndL dst src));
+  match(Set dst (AndL src mask));
 
-  format %{ "movzbq  $dst, $dst\t# long & 0xFF" %}
+  format %{ "movzbl  $dst, $src\t# long & 0xFF" %}
   ins_encode %{
-    __ movzbq($dst$$Register, $dst$$Register);
+    __ movzbl($dst$$Register, $src$$Register);
```

Comment:
Good to add a comment here that the upper 32 bits are zeroed by the instruction.